### PR TITLE
docs: Improved the Documentation on BrowserWindow type attribute

### LINKED ISSUE #34499: [Feature Request]: Improve Documentation on BrowserWindow `type` attribute
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

The docs regarding the `type` property of the BrowserWindow lacks explanation:

https://www.electronjs.org/de/docs/latest/api/browser-window

```
The possible values and behaviors of the type option are platform dependent. Mögliche Werte sind:

On Linux, possible types are desktop, dock, toolbar, splash, notification.
On macOS, possible types are desktop, textured.
The textured type adds metal gradient appearance (NSTexturedBackgroundWindowMask).
The desktop type places the window at the desktop background window level (kCGDesktopWindowLevel - 1). Note that desktop window will not receive focus, keyboard or mouse events, but you can use globalShortcut to receive input sparingly.
On Windows, possible type is toolbar.
```

It is not clear what the effects of the seperate options are, e.g.

A frameless window on linux with type `splash` is not draggable, even if the style property on the body contains `-webkit-app-region: drag`

### Proposed Solution

Add some context for the values e.g.

On Linux, possible types are:
 - desktop: ...
 -  dock: ...
 -  toolbar: ...
 -  splash: Window type `splash` does x,y and prevents the window from being draged
 -  notification: ...
 

### Alternatives Considered

None

### Additional Information

I have chosen "splash" as type of my BrowserWindows, for a reason i cant remember. On my windows machine i have been able to move the window, on linux not.

The search for the reason have more or less been "try and error" -> comment everything until it works

After i found the culprint, window type `splash`, i wanted to check what are the side effects if i remove this setting but the docs didnt have any information.

### PR DESCRIPTION
**Pull request by varunsood2003**
#### Description of Change

Closes #34499

Made changes in docs regarding the type property of the BrowserWindow that lacked explanation. Therefore added the required explanation and maintained consistency.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation, tutorials, templates, and examples are changed or added

#### Release Notes

Notes: none

### CODE REVIEW

**Review by dsanders11 — CHANGES_REQUESTED** [review]
Hi @varunsood2003! Thanks for the PR. This PR description needs to follow our [pull request template](https://raw.githubusercontent.com/electron/electron/main/.github/PULL_REQUEST_TEMPLATE.md) so that our automation around release notes works, so please revise the description to follow that. This PR also needs a more descriptive title.

For consistency, could the descriptions be changed to be more similar to the ones below them in the macOS section, namely in the style "The `<foo>` type ..." rather than "<foo>: This ..."?

**Comment by codebytere** [comment]
@varunsood2003 do you plan to pursue this PR?

**Comment by varunsood2003** [comment]
@dsanders11 I have made the required changes for this PR.

File: docs/api/structures/browser-window-options.md [hunk 1]
```diff
@@ -140,6 +140,16 @@ Possible values are:
 
 * On Linux, possible types are `desktop`, `dock`, `toolbar`, `splash`,
   `notification`.
+  * The `desktop` type places the window at the desktop background window level
+    (kCGDesktopWindowLevel - 1). However, note that a desktop window will not
+    receive focus, keyboard, or mouse events. You can still use globalShortcut to
+    receive input sparingly.
+  * The `dock` type creates a dock-like window behavior.
+  * The `Toolbar` type creates a window with a toolbar appearance.
```
Comment (dsanders11) [inline docs/api/structures/browser-window-options.md:148]:
```suggestion
  * The `toolbar` type creates a window with a toolbar appearance.
```

**Review by dsanders11 — COMMENTED** [review]
@varunsood2003, thanks! One requested change in the content, and the description on this PR still needs to be updated to fully follow the template as previously mentioned.

**Comment by varunsood2003** [comment]
Hey @dsanders11 Hopefully, I have made all the changes according to the requirements. If their are any other changes required please let me know.

**Review by dsanders11 — APPROVED** [review]

**Comment by dsanders11** [comment]
Thanks @varunsood2003! I've tweaked the PR description (release notes are not needed for documentation changes) and overall things LGTM now. 👍

cc @electron/docs

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**